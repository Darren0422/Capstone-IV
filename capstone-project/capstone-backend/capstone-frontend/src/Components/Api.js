import React, { useState } from "react";
import SearchDetails from "../Modules/SearchDetails";

function Api() {
  const [inputUsername, setInputUsername] = useState("");
  const [searchUsername, setSearchUsername] = useState(null);

  // Function to update the inputUsername
  const handleInputChange = (e) => {
    setInputUsername(e.target.value);
  };

  // Function to update the searchUsername when the form is submitted
  const handleSearch = (e) => {
    e.preventDefault();

    setSearchUsername(inputUsername);
  };

  // Passess searchUsername as props to SearchDetails module
  return (
    <div>
      <div className="App-api-container">
        <div className="App-api-search">
          <h2>Search:</h2>
          <form onSubmit={handleSearch}>
            <input
              type="text"
              name="search"
              placeholder="Username"
              value={inputUsername}
              required
              onChange={handleInputChange}
            />
            <button type="submit">Search</button>
          </form>
          <div className="App-api-searchView">
            <SearchDetails searchUsername={searchUsername} />
          </div>
        </div>
      </div>
    </div>
  );
}

export default Api;
