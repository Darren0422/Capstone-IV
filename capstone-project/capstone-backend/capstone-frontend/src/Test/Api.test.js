import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter
import Api from '../Components/Api';

// Snapshot test
test('Api component renders correctly', () => {
  const tree = renderer.create(
    <MemoryRouter>
      <Api />
    </MemoryRouter>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

