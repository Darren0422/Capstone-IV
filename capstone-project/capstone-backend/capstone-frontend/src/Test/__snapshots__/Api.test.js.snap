// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Api component renders correctly 1`] = `
<div>
  <div
    className="App-api-container"
  >
    <div
      className="App-api-search"
    >
      <h2>
        Search:
      </h2>
      <form
        onSubmit={[Function]}
      >
        <input
          name="search"
          onChange={[Function]}
          placeholder="Username"
          required={true}
          type="text"
          value=""
        />
        <button
          type="submit"
        >
          Search
        </button>
      </form>
      <div
        className="App-api-searchView"
      />
    </div>
  </div>
</div>
`;
